llvm/test/CodeGen/X86/misched-critical-path.ll

```diff
@@ -0,0 +1,234 @@
+  %pair106 = cmpxchg ptr @us, i8 %v104, i8 %v105 monotonic monotonic
+  %v106 = extractvalue { i8, i1 } %pair106, 0
+  %v107 = icmp eq i8 %v106, %v104
+  %v108 = zext i1 %v107 to i8
+  %v109 = zext i8 %v108 to i32
+  store i32 %v109, ptr @ui, align 4
+  %v110 = load i8, ptr @sc, align 1
+  %v111 = sext i8 %v110 to i32
+  %v112 = load i8, ptr @uc, align 1
+  %v113 = zext i8 %v112 to i32
+  %v114 = trunc i32 %v113 to i8
+  %v115 = trunc i32 %v111 to i8
+  %pair116 = cmpxchg ptr @si, i8 %v114, i8 %v115 monotonic monotonic
+  %v116 = extractvalue { i8, i1 } %pair116, 0
+  %v117 = icmp eq i8 %v116, %v114
+  %v118 = zext i1 %v117 to i8
+  %v119 = zext i8 %v118 to i32
+  store i32 %v119, ptr @ui, align 4
+  %v120 = load i8, ptr @sc, align 1
+  %v121 = sext i8 %v120 to i32
+  %v122 = load i8, ptr @uc, align 1
+  %v123 = zext i8 %v122 to i32
+  %v124 = trunc i32 %v123 to i8
+  %v125 = trunc i32 %v121 to i8
+  %pair126 = cmpxchg ptr @ui, i8 %v124, i8 %v125 monotonic monotonic
+  %v126 = extractvalue { i8, i1 } %pair126, 0
+  %v127 = icmp eq i8 %v126, %v124
+  %v128 = zext i1 %v127 to i8
+  %v129 = zext i8 %v128 to i32
+  store i32 %v129, ptr @ui, align 4
+  %v130 = load i8, ptr @sc, align 1
+  %v131 = sext i8 %v130 to i64
+  %v132 = load i8, ptr @uc, align 1
+  %v133 = zext i8 %v132 to i64
+  %v134 = trunc i64 %v133 to i8
+  %v135 = trunc i64 %v131 to i8
+  %pair136 = cmpxchg ptr @sl, i8 %v134, i8 %v135 monotonic monotonic
+  %v136 = extractvalue { i8, i1 } %pair136, 0
+  %v137 = icmp eq i8 %v136, %v134
+  %v138 = zext i1 %v137 to i8
+  %v139 = zext i8 %v138 to i32
+  store i32 %v139, ptr @ui, align 4
+  %v140 = load i8, ptr @sc, align 1
+  %v141 = sext i8 %v140 to i64
+  %v142 = load i8, ptr @uc, align 1
+  %v143 = zext i8 %v142 to i64
+  %v144 = trunc i64 %v143 to i8
+  %v145 = trunc i64 %v141 to i8
+  %pair146 = cmpxchg ptr @ul, i8 %v144, i8 %v145 monotonic monotonic
+  %v146 = extractvalue { i8, i1 } %pair146, 0
+  %v147 = icmp eq i8 %v146, %v144
+  %v148 = zext i1 %v147 to i8
+  %v149 = zext i8 %v148 to i32
+  store i32 %v149, ptr @ui, align 4
+  %v150 = load i8, ptr @sc, align 1
+  %v151 = sext i8 %v150 to i64
+  %v152 = load i8, ptr @uc, align 1
+  %v153 = zext i8 %v152 to i64
+  %v154 = trunc i64 %v153 to i8
+  %v155 = trunc i64 %v151 to i8
+  %pair156 = cmpxchg ptr @sll, i8 %v154, i8 %v155 monotonic monotonic
+  %v156 = extractvalue { i8, i1 } %pair156, 0
+  %v157 = icmp eq i8 %v156, %v154
+  %v158 = zext i1 %v157 to i8
+  %v159 = zext i8 %v158 to i32
+  store i32 %v159, ptr @ui, align 4
+  %v160 = load i8, ptr @sc, align 1
+  %v161 = sext i8 %v160 to i64
+  %v162 = load i8, ptr @uc, align 1
+  %v163 = zext i8 %v162 to i64
+  %v164 = trunc i64 %v163 to i8
+  %v165 = trunc i64 %v161 to i8
+  %pair166 = cmpxchg ptr @ull, i8 %v164, i8 %v165 monotonic monotonic
+  %v166 = extractvalue { i8, i1 } %pair166, 0
+  %v167 = icmp eq i8 %v166, %v164
+  %v168 = zext i1 %v167 to i8
+  %v169 = zext i8 %v168 to i32
+  store i32 %v169, ptr @ui, align 4
```

Comment:
Can you reduce this at all? It's a rather long test for this

Reply:
I did that as you suggested. Now the test really looks very refreshing. Thank you very much, @arsenm .